shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API/Validation.hs

```diff
@@ -63,9 +68,11 @@ class
     BaseM (Core.EraRule "BBODY" era) ~ ShelleyBase,
     Environment (Core.EraRule "BBODY" era) ~ STS.BbodyEnv era,
     State (Core.EraRule "BBODY" era) ~ STS.BbodyState era,
-    Signal (Core.EraRule "BBODY" era) ~ Block era
+    Signal (Core.EraRule "BBODY" era) ~ Block era,
+    HasField "pparamsUE" (utxoenv era) (Core.PParams era),
```

Comment:
Why not simply `HasField "pparamsUE" (Environment (Core.EraRule "UTXO" era)) (Core.PParams era)`? That avoids the need for this extra param.

Reply:
great!